ansible/tasks/prosody.yml

```diff
@@ -10,13 +10,10 @@
     url: "https://packages.prosody.im/debian/pubkey.asc"
 - name: "Add Prosody package repo"
   apt_repository:
-    repo: "deb https://packages.prosody.im/debian buster main"
-- name: "Detect dpkg architecture name"
-  shell: dpkg --print-architecture
-  register: dpkg_arch
+    repo: "deb https://packages.prosody.im/debian buster/snapshots/{{ prosody.snapshot }} main"
```

Comment:
Just in theory: Will this cause a new repository to be added every time the snapshot changes? How does `apt_repository` behave there?

(I know that it's irrelevant because the image is built cleanly every time)

Reply:
Good point. The `filename` key seems to be the solution: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_repository_module.html#parameter-filename

It doesn't matter due to rebuilding from scratch every time, but it would probably be nice to have it in a predictable location for debugging purposes and whatever.

Reply:
So that's an orthogonal thing Someone™ could do in a separate PR?